packages/fxa-settings/src/pages/Signin/Signin/index.tsx

```diff
@@ -0,0 +1,201 @@
+      // TODO: add message in Alert Bar re: success, then navigate to where appropriate
+    } catch (e) {
+      // TODO: metrics event for error
+      // TODO: Add in localized Alert Bar message for error
+      setError(e);
+    }
+  }, []);
+
+  const [error, setError] = useState('');
+  const [password, setPassword] = useState('');
+  const useLoggedIn = () => {};
+
+  const signInWithPassword = () => {};
+
+  // TODO:
+  // Add in the Alert Bar in place of the original `success` and `error` display divs
+
+  return (
+    <>
+      <div className="mb-4">
+        {isPasswordNeeded ? (
+          <FtlMsg id="signin-password-needed-header">
+            <h1 className="card-header">
+              Enter your password{' '}
+              <span className="card-subheader">for your Firefox account</span>
+            </h1>
+          </FtlMsg>
+        ) : (
+          <>
+            <FtlMsg id="signin">
+              <h1 className="card-header">Sign in</h1>
+            </FtlMsg>
+            <>
+              {serviceLogo ? (
+                <span className="card-subheader">
+                  <FtlMsg id="signin-subheader-with-logo">Continue to </FtlMsg>
+                  {serviceLogo}
+                </span>
+              ) : (
+                <FtlMsg
+                  id="signin-subheader-without-logo"
+                  vars={{ serviceName }}
+                >
+                  <span className="card-subheader">
+                    Continue to {serviceName}
+                  </span>
+                </FtlMsg>
+              )}
+            </>
+          </>
+        )}
+      </div>
+      <section>
+        {/* Alerts and success messages originally went here */}
+        <div className="mt-9">
+          <div className="avatar-wrapper mx-auto h-24 w-24 mobileLandscape:h-40 mobileLandscape:w-40"></div>
+          <div className="my-5 text-base break-all">{email}</div>
+        </div>
+        <form noValidate {...{ onSubmit }}>
+          <input type="email" className="email hidden" value={email} disabled />
+
+          {isPasswordNeeded && (
+            <InputPassword
+              className="mb-2"
+              label={localizedPasswordFormLabel}
+              hasErrors={error.length > 0}
+              errorText={error}
+              tooltipPosition="bottom"
+              required={true}
+              autoFocus={true}
+              onChange={(e) => setPassword(e.currentTarget.value)}
+            />
+          )}
+          {/* This non-fulfilled input tricks the browser, when trying to
+              sign in with the wrong password, into not showing the doorhanger.
+           */}
+          <input className="hidden" required />
+
+          <div className="flex">
+            <button
+              className="cta-primary cta-xl"
+              type="submit"
+              onClick={isPasswordNeeded ? useLoggedIn : signInWithPassword}
+            >
+              <FtlMsg id="signin">Sign in</FtlMsg>
```

Comment:
Should `<FtlMsg>` wrap the `<button>`?

@bcolsson Should the FTL string have a more detailed ID that includes the component name to avoid potential conflicts with other strings? Something like `signin-submit-button` instead of `signin`? Asking for my personal knowledge too!

Reply:
This is a great question re: the ID, and I also want to know! I figured it might be helpful to have it be less specific so that it can be reused without being confusing? But maybe that isn't how it works?

Reply:
(Also I am still learning about `<FtlMsg>` so I'm not sure, but I think you're right!

Reply:
@vpomerleau / @millsoper - Oops missed this comment here, but yes I left a comment below along those lines asking for a new ID. Avoiding conflicts is great and necessary, and using the component ID is one strategy to avoid that. From an l10n point of view, avoiding conflicts with other strings is something we'd hope the person generating the source string IDs is checking.

The more important aspect for l10n, is that the string IDs often serve as context markers. In this case "Sign in" could be a noun or a verb, a title or a suggestion to do something. Adding the context that this is a button allows languages to determine how to localize this appropriately for their locale.